src/finiteVolume/processes/les.cpp

```diff
@@ -0,0 +1,252 @@
+
+ablate::finiteVolume::processes::LES::LES(std::shared_ptr<eos::transport::TransportModel> transportModelIn) : transportModel(std::move(transportModelIn)) {}
+
+void ablate::finiteVolume::processes::LES::Setup(ablate::finiteVolume::FiniteVolumeSolver& flow) {
+    if (transportModel) {
+        if (diffusionData.diffFunction.function) {
+            // Register the euler/Momentum LESdiffusion source term
+            flow.RegisterRHSFunction(LESMomentumFlux,
+                                     &diffusionData,
+                                     CompressibleFlowFields::EULER_FIELD,
+                                     {CompressibleFlowFields::EULER_FIELD, CompressibleFlowFields::DENSITY_EV_FIELD},
+                                     {CompressibleFlowFields::VELOCITY_FIELD});
+            // Register the euler/Energy LESdiffusion source term
+            flow.RegisterRHSFunction(LESEnergyFlux,
+                                     &diffusionData,
+                                     CompressibleFlowFields::EULER_FIELD,
+                                     {CompressibleFlowFields::EULER_FIELD, CompressibleFlowFields::DENSITY_EV_FIELD},
+                                     {CompressibleFlowFields::TEMPERATURE_FIELD, CompressibleFlowFields::VELOCITY_FIELD});
+            // Register the Species LESdiffusion source term
+            flow.RegisterRHSFunction(LESSpeciesFlux,
+                                     &diffusionData,
+                                     CompressibleFlowFields::DENSITY_YI_FIELD,
+                                     {CompressibleFlowFields::DENSITY_YI_FIELD, CompressibleFlowFields::DENSITY_EV_FIELD},
+                                     {CompressibleFlowFields::EV_FIELD});
+            // Register the ev LESdiffusion source term
+            flow.RegisterRHSFunction(
+                LESevFlux, &diffusionData, CompressibleFlowFields::EV_FIELD, {CompressibleFlowFields::EULER_FIELD, CompressibleFlowFields::DENSITY_EV_FIELD}, {CompressibleFlowFields::EV_FIELD});
+        }
+    }
+
+    diffusionData.computeTemperatureFunction = eos->GetThermodynamicFunction(eos::ThermodynamicProperty::Temperature, flow.GetSubDomain().GetFields());
+}
+
+PetscErrorCode ablate::finiteVolume::processes::LES::LESMomentumFlux(PetscInt dim, const PetscFVFaceGeom* fg, const PetscInt* uOff, const PetscInt* uOff_x, const PetscScalar* field,
+                                                                     const PetscScalar* grad, const PetscInt* aOff, const PetscInt* aOff_x, const PetscScalar* aux, const PetscScalar* gradAux,
+                                                                     PetscScalar* flux, void* ctx) {
+    PetscFunctionBeginUser;
+    const int VEL = 1;
+    //?? RHOU initialized here??//
+    for (PetscInt d = 0; d < dim; d++) {
+        flux[CompressibleFlowFields::RHOU + d] = 0.0;
+    }
+    // Compute the LES stress tensor tau
+    PetscReal LESTau[9];  // Maximum size without symmetry
```

Comment:
No, the source term for the momentum terms is set to zero because it is += later.